hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/EventService.java

```diff
@@ -72,26 +73,23 @@ EventRegistration registerLocalListener(@Nonnull String serviceName,
      * @param topic       topic name
      * @param filter      event filter
      * @param listener    listener instance
-     * @return event registration
+     * @return event registration future
      * @throws IllegalArgumentException if the listener or filter is {@code null}
      */
-    EventRegistration registerLocalListener(@Nonnull String serviceName,
-                                            @Nonnull String topic,
-                                            @Nonnull EventFilter filter,
-                                            @Nonnull Object listener);
+    CompletableFuture<EventRegistration> registerLocalListener(@Nonnull String serviceName, @Nonnull String topic,
+                                                               @Nonnull EventFilter filter, @Nonnull Object listener);
 
     /**
      * Registers a listener on all cluster nodes.
      *
      * @param serviceName service name
      * @param topic       topic name
      * @param listener    listener instance
-     * @return event registration
+     * @return event registration future
      * @throws IllegalArgumentException if the listener is {@code null}
      */
-    EventRegistration registerListener(@Nonnull String serviceName,
-                                       @Nonnull String topic,
-                                       @Nonnull Object listener);
+    CompletableFuture<EventRegistration> registerListener(@Nonnull String serviceName, @Nonnull String topic,
```

Comment:
what about introducing an async variant besides sync one? This can lead less change and make the places which we need async calls clear.